cartoframes/data/dataset.py

```diff
@@ -662,16 +710,12 @@ def get_table_columns(self):
             '''.format(table=self._table_name, schema=self._schema)
 
             try:
-                table_info = self._con.sql_client.send(query)
+                table_info = self._client.execute_query(query)
                 return [Column(c['column_name'], pgtype=c['data_type']) for c in table_info['rows']]
             except CartoException as e:
                 # this may happen when using the default_public API key
                 if str(e) == 'Access denied':
```

Comment:
For a future refactor, we should be able to catch [this error instead](https://github.com/danicarrion/pyrestcli/blob/master/pyrestcli/exceptions.py#L45)

Reply:
:+1:

Reply:
https://github.com/CartoDB/cartoframes/issues/836